### PR TITLE
Zc/iteration 3.5 backend

### PR DESCRIPTION
**Pull request by zclewell**
## Status: 
:rocket: Ready

## Description
Made tests for validate actually do something

### CODE REVIEW

File: backend/validation.py [hunk 1]
```diff
@@ -1,25 +1,45 @@
...
+    def __init__(self, message):
+        super().__init__(message)
+
+
+class UpcValidationException(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+INCORRECT_FIELD_ERROR = 'Incorrect Amount of Fields'
```
Comment (rsun0) [inline backend/validation.py]: You could place these constants within their relevant classes

File: backend/tests/test_validation.py [hunk 2]
```diff
@@ -34,19 +32,17 @@ def test_invalid_location_too_many_fields():
         'extra': 'peilun'
     }
     try:
-        validation.validate_location(loc)
-        assert False
-    except:
-        pass
+        validate_location(loc)
```
Comment (rsun0) [inline backend/tests/test_validation.py:35]: All invalid input test cases should fail if validate() does not throw

**Review by rsun0 — CHANGES_REQUESTED** [review]
Please make the invalid test cases fail when validation incorrectly passes. Also, optionally move string constants into their corresponding classes.

File: backend/tests/test_validation.py (hunk 2 above)
Comment (zclewell) [reply backend/tests/test_validation.py:35]: Corrected

File: backend/validation.py (hunk 1 above)
Comment (zclewell) [reply backend/validation.py]: Corrected

File: backend/validation.py [hunk 3]
```diff
@@ -1,25 +1,48 @@
...
+
+class UpcValidationException(Exception):
+    NOT_DIGITS_ERROR = 'Upc must be digits'
+    INCORRECT_LENGTH_ERROR = 'Upc must have length 12'
+
+    def __init__(self, message):
+        super().__init__(message)
+
+
+LocationValidationException.INVALID_LONG_VALUE_ERROR
```
Comment (rsun0) [inline backend/validation.py]: Was this supposed to be deleted?

**Review by rsun0 — CHANGES_REQUESTED** [review]

File: backend/validation.py (hunk 3 above)
Comment (zclewell) [reply backend/validation.py]: Yes, removed

**Review by rsun0 — APPROVED** [review]
Thanks for your changes

**Pull request merged**